core/src/listener/listener.ts

```diff
@@ -329,30 +346,39 @@ function historyJob({
 function processEventJob({
   workerQueue,
   processFn,
+  service,
   logger
 }: {
   workerQueue: Queue
   processFn: (log: ethers.Event) => Promise<ProcessEventOutputType | undefined>
+  service: string
   logger: Logger
 }) {
   const _logger = logger.child({ name: 'processEventJob', file: FILE_NAME })
 
   async function wrapper(job: Job) {
     const inData: IProcessEventListenerJob = job.data
-    const { event } = inData
-    _logger.debug(event, 'event')
+    const { events, blockNumber } = inData
 
     try {
-      const jobMetadata = await processFn(event)
-      if (jobMetadata) {
-        const { jobId, jobName, jobData, jobQueueSettings } = jobMetadata
-        const queueSettings = jobQueueSettings ? jobQueueSettings : LISTENER_JOB_SETTINGS
-        await workerQueue.add(jobName, jobData, {
-          jobId,
-          ...queueSettings
-        })
-        _logger.debug(`Listener submitted job [${jobId}] for [${jobName}]`)
+      for (const event of events) {
+        _logger.debug(event, 'event')
+        const jobMetadata = await processFn(event)
+        if (jobMetadata) {
```

Comment:
@Intizar-T Can you please add `else` branch and log.error 🙏

Reply:
should I also throw the error so that the processEventJob can retry it? because `processFn()` tries to access to the db it might worth a retry in case if the error is due to querying the db?

Reply:
Sounds good!